### PR TITLE
feat: speed up unmarshaller with common signature trees

### PR DESCRIPTION
**Pull request by bdraco**
This is ~4%-6% speed up (there is some margin of error)

### CODE REVIEW

File: src/dbus_fast/_private/unmarshaller.py [hunk 1]
```diff
@@ -50,6 +50,13 @@
 HEADER_ARRAY_OF_STRUCT_SIGNATURE_POSITION = 12
 
 
+SIGNATURE_TREE_EMPTY = get_signature_tree("")
+SIGNATURE_TREE_N = get_signature_tree("n")
+SIGNATURE_TREE_S = get_signature_tree("s")
+SIGNATURE_TREE_SA_SV_AS = get_signature_tree("sa{sv}as")
+SIGNATURE_TREE_SA_SV_AS_TYPES_1 = SIGNATURE_TREE_SA_SV_AS.types[1]
+SIGNATURE_TREE_SA_SV_AS_TYPES_2 = SIGNATURE_TREE_SA_SV_AS.types[1]
```
Comment (dlech) [inline src/dbus_fast/_private/unmarshaller.py:58]: typo?

```suggestion
SIGNATURE_TREE_SA_SV_AS_TYPES_2 = SIGNATURE_TREE_SA_SV_AS.types[2]
```

Comment (bdraco) [reply src/dbus_fast/_private/unmarshaller.py]: Yes. Hopefully why the test didn't pass

Comment (bdraco) [reply src/dbus_fast/_private/unmarshaller.py]: Thanks!

**Comment by bdraco** [comment]
Production testing 👍

**Pull request merged**